test/user.test.js

```diff
@@ -1404,6 +1425,92 @@ describe('User', function() {
     }
   });
 
+  describe('User.setPassword()', () => {
+    let userId;
+    beforeEach(givenUserId);
+
+    it('changes the password - callback-style', done => {
+      User.setPassword(userId, 'new password', (err) => {
+        if (err) return done(err);
+        expect(arguments.length, 'changePassword callback arguments length')
+          .to.be.at.most(1);
+
+        User.findById(userId, (err, user) => {
+          if (err) return done(err);
+          user.hasPassword('new password', (err, isMatch) => {
+            if (err) return done(err);
+            expect(isMatch, 'user has new password').to.be.true();
+            done();
+          });
+        });
+      });
+    });
+
+    it('changes the password - Promise-style', () => {
+      return User.setPassword(userId, 'new password')
+        .then(() => {
+          expect(arguments.length, 'changePassword promise resolution')
+            .to.equal(0);
+          return User.findById(userId);
+        })
+        .then(user => {
+          return user.hasPassword('new password');
+        })
+        .then(isMatch => {
+          expect(isMatch, 'user has new password').to.be.true();
+        });
+    });
+
+    it('fails with 401 for unknown user', () => {
+      return User.setPassword('unknown-id', 'pass').then(
+        success => { throw new Error('setPassword should have failed'); },
+        err => {
+          // workaround for chai problem
+          //   object tested must be an array, an object, or a string,
+          //   but error given
+          const props = Object.assign({}, err);
+          expect(props).to.contain({
+            code: 'USER_NOT_FOUND',
+            statusCode: 401,
+          });
+        });
+    });
+
+    it('forwards the "options" argument', () => {
```

Comment:
I think I found the other comment: https://github.com/strongloop/loopback/pull/3350#discussion_r110979869

Let me explain why I think we should keep using both "forwards" and "injects":

The test `it forwards the "options" argument` verifies that the implementation of `User.setPassword` is forwarding the`options` argument whenever calling other User and DAO methods, to ensure that context can be correctly propagated.

The second test, "it injects reset password options from remoting context" verifies that remoting metadata of `setPassword` provides correct configuration for the `options` argument, a configuration instructing loopback/strong-remoting to inject options built from the request context.

In my mind, these two test cases are different, and therefore it makes sense to me to use different verbs to describe them.